src/utils/Validator.ts

```diff
@@ -70,4 +74,107 @@ export class Validator {
       }
     });
   }
+
+  /**
+   * Validates metric name.
+   * @see [CloudWatch Metric](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_MetricDatum.html)
+   *
+   * @param key
+   * @param value
+   *
+   * @throws {InvalidMetricError} Metric name must be valid.
+   */
+  public static validateMetric(key: string, value: number, unit?: Unit | string): void {
+    if (key.trim().length == 0) {
+      throw new InvalidMetricError(`Metric key ${key} must include at least one non-whitespace character`);
+    }
+
+    if (key.length > Constants.MAX_METRIC_NAME_LENGTH) {
+      throw new InvalidMetricError(
+        `Metric key ${key} must not exceed maximum length ${Constants.MAX_METRIC_NAME_LENGTH}`,
+      );
+    }
+
+    if (isNaN(value) || value === -Infinity || value === Infinity) {
```

Comment:
Can we use `Number.isNaN` here? `isNaN` has some weird behaviour: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN#confusing_special-case_behavior

Reply:
Lets add some more `putMetric` validation tests as well to some of these examples of weird input/behaviour.

Reply:
The only weird inputs I can give are `NaN` and `Infinity` given that the value must be of type `number`.